### PR TITLE
feat(hyd): A380 auto brakes

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #[issue_no]

## Summary of Changes

First implementation of auto-brakes on the A380.

Revised logic

- [x] RTO button + selection knob
- [x] Activation after 5s or nose compressed
- [x] Deactivation from auto throttle instinctive disconnect (except RTO)
- [x] DECEL light A380 hysteresis values
- [x] Correct deceleration for each mode
- [x] DC Solenoid powered knob for return to disarm position
- [x] Knob animation with fast return to disarm position when solenoid is powered

Placeholder BTV delaying braking for less runway occupancy time. Full implementation later.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Check RTO on takeoff
All modes on landing
Should react to SET events
Should deactivate on auto throttle disconnect (except in RTO)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs
Comment (2hwk) [inline fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs:4187]: Assuming these are intentional changes (A320)?

File: fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs
Comment (2hwk) [inline fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs:4217]: Assuming these are intentional changes (A320)?

**Review by 2hwk — CHANGES_REQUESTED** [review]

**Comment by 2hwk** [comment]
Merge conflicts?

**Comment by 2hwk** [comment]
image

It never ends 😭

**Comment by flogross89** [comment]
[autobrk-patch.diff.txt](https://github.com/flybywiresim/aircraft/files/14391573/autobrk-patch.diff.txt)
FMA for RTO was wrong, this fixes it (see discord). The merge conflicts come from my tooltips in the XML, we should just accept the combination of both there

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/en-US.locPak
Comment (Benjozork) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/en-US.locPak:18]: **suggestion:** prefix those with `A380X`

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/en-US.locPak
Comment (Benjozork) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/en-US.locPak:19]:
```suggestion
      "COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_HDG_TUNING_RIGHT": "Adjust selected heading to the right (%((A:AUTOPILOT HEADING LOCK DIR:1,degrees))%!d!°)",
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: fbw-a380x/src/wasm/fbw_a380/src/interface/SimConnectInterface.cpp
Comment (Gurgel100) [inline fbw-a380x/src/wasm/fbw_a380/src/interface/SimConnectInterface.cpp:2324]:
```suggestion
      execute_calculator_code("(>K:A32NX_AUTO_THROTTLE_DISCONNECT)", nullptr, nullptr, nullptr);
```

Comment (lukecologne) [reply fbw-a380x/src/wasm/fbw_a380/src/interface/SimConnectInterface.cpp:2324]: Periods are required in custom event names according to the MSFS SDK docs

Comment (Gurgel100) [reply fbw-a380x/src/wasm/fbw_a380/src/interface/SimConnectInterface.cpp:2324]: Didn't know that. Thanks for the heads up.

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/SimObjects/AirPlanes/FlyByWire_A380_842/model/behaviour/mip.xml
Comment (frankkopp) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/SimObjects/AirPlanes/FlyByWire_A380_842/model/behaviour/mip.xml:111]: I think this should be a momentary as it does not stay "in" (it is not a switch) - so it either resets itself to 0 or the systems code should. 
Otherwise after pressing it via API the user must reset it again so it is not "held" all the time.

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/de-DE.locPak
Comment (frankkopp) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/de-DE.locPak:10]: you shouldn't even have these files in the base folder any more!?

File: fbw-a380x/src/localization/msfs/de-DE.locPak
Comment (frankkopp) [inline fbw-a380x/src/localization/msfs/de-DE.locPak:18]: You only have to do the en-US.locPak - the rest is handled on Localazy. It downloads the latest language files every build.
So if you add a key to the en-US.locPak we will upload it to Localazy and it will automatically be added to all languages. The translation then also happens on Localazy.

File: fbw-a380x/src/wasm/extra-backend-a380x/src/AircraftPresets/AircraftPresetProcedures_A380X.h
Comment (frankkopp) [inline fbw-a380x/src/wasm/extra-backend-a380x/src/AircraftPresets/AircraftPresetProcedures_A380X.h:168]: This is not correct in how you setup the the vars - you would basically hold the button all the time (it is not a switch). See remark above. 

Correct with your current button logic would be:
```cpp
        ProcedureStep{"Autobrake RTO",        3068, false,  200,  "(L:A32NX_AUTOBRAKES_RTO_ARMED) 1 ==",                "1 (>L:A32NX_OVHD_AUTOBRK_RTO_ARM_IS_PRESSED)"},
        ProcedureStep{"Autobrake RTO",        3068, false, 2000,  "(L:A32NX_OVHD_AUTOBRK_RTO_ARM_IS_PRESSED) 0 ==",     "0 (>L:A32NX_OVHD_AUTOBRK_RTO_ARM_IS_PRESSED)"},
```
Not very nice as one has to manually reset the "pressed" push button to 0:

File: fbw-a380x/src/wasm/fbw_a380/src/interface/SimConnectInterface.cpp
Comment (frankkopp) [reply fbw-a380x/src/wasm/fbw_a380/src/interface/SimConnectInterface.cpp:2324]: This looks strange - why would the AUTO_THROTTLE_DISCONNECT itself re-emmit itself - wouldn't that cause a loop?

File: fbw-a380x/src/wasm/systems/a380_systems_wasm/src/autobrakes.rs
Comment (frankkopp) [inline fbw-a380x/src/wasm/systems/a380_systems_wasm/src/autobrakes.rs:32]: Why is this different from LO and MED?

**Review by frankkopp — CHANGES_REQUESTED** [review]
Please document you vars here: 
`fbw-a380x/docs/a380-simvars.md`

Also you only have to change this locPak file: 
`fbw-a380x/docs/a380-simvars.md`
No other has to be touched. Read here:
`fbw-a380x/src/localization/README.md`

All in all a bit more documentation about how it is supposed to work in regards of the vars and events wouldn't hurt - maybe in the rs file?

File: fbw-a380x/src/wasm/systems/a380_systems_wasm/src/autobrakes.rs
Comment (tracernz) [reply fbw-a380x/src/wasm/systems/a380_systems_wasm/src/autobrakes.rs:32]: > why would the AUTO_THROTTLE_DISCONNECT itself re-emmit itself

It's a different event, for use by our other systems. The FBW interface masks the sim A/T disconnect event so nobody else can listen for it.

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/de-DE.locPak
Comment (frankkopp) [reply fbw-a380x/src/base/flybywire-aircraft-a380-842/de-DE.locPak]: > Those came after a merge conflict.... If there's still a file diff in here it's that file is still on master isn't it? Should we remove those from master first maybe?

they are not on master any more - I removed them there - see screenshot

https://github.com/flybywiresim/aircraft/tree/master/fbw-a380x/src/base/flybywire-aircraft-a380-842
image

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/SimObjects/AirPlanes/FlyByWire_A380_842/model/behaviour/mip.xml
Comment (frankkopp) [reply fbw-a380x/src/base/flybywire-aircraft-a380-842/SimObjects/AirPlanes/FlyByWire_A380_842/model/behaviour/mip.xml:111]: Working as expected indeed - resolved.
https://github.com/flybywiresim/aircraft/assets/16833201/07e17c32-9b21-4352-916a-31947d8c16cb

**Review by 2hwk — APPROVED** [review]

**Review by frankkopp — APPROVED** [review]

File: fbw-a380x/src/wasm/extra-backend-a380x/src/AircraftPresets/AircraftPresetProcedures_A380X.h
Comment (frankkopp) [reply fbw-a380x/src/wasm/extra-backend-a380x/src/AircraftPresets/AircraftPresetProcedures_A380X.h]: see above - resolved

**Pull request merged**
